vector/src/main/java/im/vector/app/features/home/room/detail/TimelineViewModel.kt

```diff
@@ -136,17 +138,22 @@ class TimelineViewModel @AssistedInject constructor(
         private val notificationDrawerManager: NotificationDrawerManager,
         private val locationSharingServiceConnection: LocationSharingServiceConnection,
         private val stopLiveLocationShareUseCase: StopLiveLocationShareUseCase,
+        private val coroutineDispatchers: CoroutineDispatchers,
         timelineFactory: TimelineFactory,
         appStateHandler: AppStateHandler,
 ) : VectorViewModel<RoomDetailViewState, RoomDetailAction, RoomDetailViewEvents>(initialState),
         Timeline.Listener, ChatEffectManager.Delegate, CallProtocolsChecker.Listener, LocationSharingServiceConnection.Callback {
 
-    private val room = session.getRoom(initialState.roomId)!!
+    private val room = viewModelScope.async(start = CoroutineStart.LAZY) {
+        session.roomService().awaitRoom(initialState.roomId)!!
```

Comment:
not for this PR, with this async I'm hoping it'll  be easier to gracefully handle when the room is null :crossed_fingers: 
[#6875 ](https://github.com/vector-im/element-android/issues/5611)